src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
+    if (openingComment) {
+      DOMChildrenOperations.removeDelimitedText(
```

Comment:
That is, I think it's better if DOMChildrenOperations is aware of this comment/text-node structure instead of only removing the closing node.